tests/storage/storage.go

```diff
@@ -140,6 +140,90 @@ var _ = SIGDescribe("Storage", func() {
 
 			runHostPathJobAndExpectCompletion(pod)
 		}
+
+		Context("with error disk", func() {
+			var (
+				nodeName, address, device string
+
+				pvc *k8sv1.PersistentVolumeClaim
+			)
+
+			BeforeEach(func() {
+				nodeName = tests.NodeNameWithHandler()
+				address, device = tests.CreateErrorDisk(nodeName)
+				var err error
+				_, pvc, err = tests.CreatePVandPVCwithFaultyDisk(nodeName, device, tests.NamespaceTestDefault)
+				Expect(err).NotTo(HaveOccurred(), "Failed to create PV and PVC for faulty disk")
+			})
+
+			AfterEach(func() {
+				tests.RemoveErrorDisk(nodeName, address)
+			})
+
+			It("should pause VMI on IO error", func() {
+				By("Creating VMI with faulty disk")
+				vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+				vmi = tests.AddPVCDisk(vmi, "pvc-disk", "virtio", pvc.Name)
+				_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+				Expect(err).To(BeNil(), "Failed to create vmi")
+
+				tests.WaitForSuccessfulVMIStartWithTimeoutIgnoreWarnings(vmi, 120)
+
+				By("Reading from disk")
+				Expect(console.LoginToAlpine(vmi)).To(Succeed())
+				Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: "nohup sh -c \"sleep 10 && while true; do dd if=/dev/vdb of=/dev/null >/dev/null 2>/dev/null; done\" & \n"},
```

Comment:
Just a minor remark. You might want to fetch the disk device from the vmi status instead of hardcoding the device

Reply:
I think it should be ok for now since we don't change the os that much. I will have a look into it for future